packages/components/src/mobile/image/index.native.js

```diff
@@ -80,6 +83,27 @@ const ImageComponent = ( {
 					onImageDataLoad( metaData );
 				}
 			} );
+
+			if ( url.startsWith( 'file:///' ) ) {
+				setLocalURL( url );
```

Comment:
My guess is that this replacing images does not immediate display the new local image or perform smooth transitions to the network image because the `networkUrl` and `networkImageLoaded` values are not reset when a new, replacement local URL is provided. I wonder if we could resolve those issues by reseting the values here. 

From my initial testing, the following improves the situation. WDYT? 

```suggestion
				setLocalURL( url );
				setNetworkURL( null );
				setNetworkImageLoaded( false );
```

Reply:
Good suggestion. I had explored this idea initially, but my approach had some unintended effects on Android. This appears to mitigate the issue for both platforms, however, so I'll commit it and do further testing for both. 🙇

Reply:
> I had explored this idea initially, but my approach had some unintended effects on Android. This appears to mitigate the issue for both platforms, however, so I'll commit it and do further testing for both. 🙇

🤔 Alright, please be sure to retest these changes to verify it does not introduce the unintended effect on Android. I am unfamiliar with what specifically that may be.